src/Dma.h

```diff
@@ -107,9 +113,11 @@ namespace RTX {
     
     // business logic
     virtual int allocateDemandToJunctions(time_t time);
+    DemandAllocationDelegate *demandAllocationDelegate;
```

Comment:
This should probably be private and then set using an appropriate setter function.

Reply:
Convention in this codebase is to use shared pointers for jobs like this to avoid memory allocation book-keeping.

```
private:
    std::shared_ptr<DemandAllocationDelegate> _allocationDelegate;
```